pkg/ctl/create/nodegroup.go

```diff
@@ -153,22 +153,10 @@ func doCreateNodeGroups(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg s
 			if cmd.Flag(f).Changed {
 				return fmt.Errorf("cannot use --%s unless a config file is specified via --config-file/-f", f)
 			}
+			return nil
 		}
 
-		err := CheckEachNodeGroup(ngFilter, cfg, func(i int, ng *api.NodeGroup) error {
-			if ng.AllowSSH && ng.SSHPublicKeyPath == "" {
-				return fmt.Errorf("--ssh-public-key must be non-empty string")
-			}
-
-			// generate nodegroup name or use either flag or argument
-			if utils.NodeGroupName(ng.Name, nameArg) == "" {
-				return cmdutils.ErrNameFlagAndArg(ng.Name, nameArg)
-			}
-			ng.Name = utils.NodeGroupName(ng.Name, nameArg)
```

Comment:
I missed this... the behaviour here is different from the other version, we will need to reinstate it somehow.

Reply:
Maybe I messed up in the rebase, but initially, I did the renaming of the ng.Name with the nameArg when it was provided. I can fix that

Reply:
No to worry, I have a fix for this in #671.